cmake/psi4OptionsTools.cmake

```diff
@@ -200,4 +200,29 @@ if(${ENABLE_${PLUGIN_NAME}})
 else()
    add_library(${plugin_name} INTERFACE)
 endif()
-endmacro(optional_plugin plugin_name)
+endmacro(optional_plugin plugin_name test_names)
+
+#Macro for adding a skeleton plugin
+macro(add_skeleton_plugin PLUG EXTRA_FLAGS)
+    set(CCSD "${CMAKE_CURRENT_SOURCE_DIR}")
+    set(CCBD "${CMAKE_CURRENT_BINARY_DIR}")
+    set(PSIEXE ${STAGED_INSTALL_PREFIX}/${CMAKE_INSTALL_BINDIR}/psi4)
+    set(DIR_2_PASS ${CMAKE_PREFIX_PATH} ${STAGED_INSTALL_PREFIX})
+add_custom_target(plugin_${PLUG}
+    ALL
+    DEPENDS psi4-core
+    COMMAND ${CMAKE_COMMAND} -E remove_directory ${CCBD}/${PLUG}
+    COMMAND ${PSIEXE} --plugin-name ${PLUG} ${EXTRA_FLAGS}
+    COMMAND ${CMAKE_COMMAND} -E chdir "${CCBD}/${PLUG}" cmake -C ${STAGED_INSTALL_PREFIX}/share/cmake/psi4/psi4PluginCache.cmake "-DCMAKE_PREFIX_PATH=${DIR_2_PASS}" .
+    COMMAND ${CMAKE_COMMAND} -E chdir "${CCBD}/${PLUG}" ${CMAKE_MAKE_PROGRAM}
+    COMMAND ${CMAKE_COMMAND} -E create_symlink ${CCBD}/${PLUG}/input.dat ${CCSD}/input.dat
+    COMMAND ${CMAKE_COMMAND} -E create_symlink "${PLUG}/${PLUG}.so" "${PLUG}.so"
+    COMMAND ${CMAKE_COMMAND} -E create_symlink "${PLUG}/__init__.py" "__init__.py"
+    COMMAND ${CMAKE_COMMAND} -E create_symlink "${PLUG}/pymodule.py" "pymodule.py"
+    COMMENT "Build ${PLUG} example plugin"
+    VERBATIM)
+include(TestingMacros)
+add_regression_test(${PLUG} "${test_names}")
+endmacro(add_skeleton_plugin PLUG EXTRA_FLAGS)
```

Comment:
Ok, nice. FYI, CMake itself has deprecated the crazy repeat-loop-opening-args-in-closing-and-else, so you needn't feel obliged to carry on with it.